### ISSUE TITLE
Rewrite TokenizedInput for cleanliness & to better control re-renders

### ISSUE THREAD

**Issue by daltojohnso** [metabase/metabase#14878]
**Description**
We should update and improve the implementation of `TokenizedInput`. Since it is an input field, performance hitches and awkward rendering behavior are more noticeable to the end user. We can likely move it to use `React.memo` + hooks to gain a better handle on the event listeners it sets up and the rate at which it re-renders.

- A lot of unnecessary re-renders occur in the `TokenizedInput` component. For example, the `style` prop as currently handled by one of its parent components causes a re-render every time the parent re-renders.
- Some of the implementation is a bit dated -- `ReactDOM.findDOMNode(this)` instead of refs, deprecated lifecycle methods, etc.

**Comment by howonlee** [comment]
ariya and I are thinking about whacking this and replacing with our copy of ace, because ace has facilities for highlighting and all the stuff that we're looking.

**Comment by daltojohnso** [comment]
Good idea, that would remove a lot of complexity that we shouldn't be concerned about anyways

**Comment by ariya** [comment]
> ariya and I are thinking about whacking this and replacing with our copy of ace, because ace has facilities for highlighting and all the stuff that we're looking.

Plus, we're already using ACE anyway for the SQL editor.
